backend/lcfs/web/api/compliance_report/services.py

```diff
@@ -39,20 +45,103 @@ async def create_compliance_report(
+            raise DataNotFoundException("Compliance period not found.")
+
         draft_status = await self.repo.get_compliance_report_status_by_desc(
             report_data.status
         )
-        report = await self.repo.add_compliance_report(
-            ComplianceReport(
-                compliance_period=period,
-                organization_id=organization_id,
-                current_status=draft_status,
-                summary=ComplianceReportSummary(),  # Create an empty summary object
-            )
+        if not draft_status:
+            raise DataNotFoundException(f"Status '{report_data.status}' not found.")
+
+        # Generate a new group_uuid for the new report series
+        group_uuid = str(uuid.uuid4())
+
+        report = ComplianceReport(
+            compliance_period_id=period.compliance_period_id,
+            organization_id=organization_id,
+            current_status_id=draft_status.compliance_report_status_id,
+            reporting_frequency=ReportingFrequency.ANNUAL,
+            compliance_report_group_uuid=group_uuid,  # New group_uuid for the series
+            version=0,  # Start with version 0
+            nickname="Original Report",
+            summary=ComplianceReportSummary(),  # Create an empty summary object
         )
+
+        # Add the new compliance report
+        report = await self.repo.add_compliance_report(report)
+
+        # Create the history record
         await self.repo.add_compliance_report_history(report, self.request.user)
 
-        return report
+        return ComplianceReportBaseSchema.model_validate(report)
+
+    @service_handler
+    async def create_supplemental_report(
+        self, report_id: int
+    ) -> ComplianceReportBaseSchema:
+        """
+        Creates a new supplemental compliance report.
+        The report_id can be any report in the series (original or supplemental).
+        Supplemental reports are only allowed if the status of the current report is 'Assessed'.
+        """
+
+        user: UserProfile = self.request.user
+
+        # Fetch the current report using the provided report_id
+        current_report = await self.repo.get_compliance_report_by_id(
+            report_id, is_model=True
+        )
+        if not current_report:
+            raise DataNotFoundException("Compliance report not found.")
+
+        # Validate that the user has permission to create a supplemental report
+        if user.organization_id != current_report.organization_id:
+            raise ServiceException(
+                "You do not have permission to create a supplemental report for this organization."
+            )
+
+        # Validate that the status of the current report is 'Assessed'
+        # if current_report.current_status.status != ComplianceReportStatusEnum.ASSESSED:
+        #     raise ServiceException(
+        #         "A supplemental report can only be created if the current report's status is 'Assessed'."
+        #     )
```

Comment:
is this commented out because the status of the compliance report can be in any status to create a supplemental report?

Reply:
good catch, this was just for testing. fixed.